app/src/split/bluetooth/central.c

```diff
@@ -33,6 +33,29 @@ static struct bt_uuid_128 uuid = BT_UUID_INIT_128(ZMK_SPLIT_BT_SERVICE_UUID);
 static struct bt_gatt_discover_params discover_params;
 static struct bt_gatt_subscribe_params subscribe_params;
 
+struct zmk_split_peripheral_event {
+    uint32_t position;
+    uint32_t state;
+    int32_t timestamp;
+};
+
+K_MSGQ_DEFINE(peripheral_event_msgq, sizeof(struct zmk_split_peripheral_event), 4, 4);
+
+void peripheral_event_work_callback(struct k_work *work) {
+    struct zmk_split_peripheral_event ev;
+    while (k_msgq_get(&peripheral_event_msgq, &ev, K_NO_WAIT) == 0) {
+        struct position_state_changed *pos_ev = new_position_state_changed();
+        pos_ev->position = ev.position;
+        pos_ev->state = ev.state;
+        pos_ev->timestamp = ev.timestamp;
```

Comment:
slightly unrelated (not something we necessarily have to pick up in this ticket) but maybe we should always use `create_position_state_changed(position, state, timestamp)` functions to create events so the compiler complains about all call sites when a new field is added?